app/controllers/spree/adyen_notifications_controller.rb

```diff
@@ -10,10 +10,14 @@ def notify
       accept and return
     end
 
-    Spree::Adyen::NotificationProcessor.new(notification).process!
+    notification.save!
 
-    # accept after processing has completed
-    accept
+    begin
+      Spree::Adyen::NotificationProcessor.new(notification).process!
+    ensure
+      # must accept notification, but still bubble error
+      accept
```

Comment:
Apparently due to the was rails renders things the page still 500's despite this clause. This is saved by the fact that the notification exists and the dupliciate clauses renders the page.